# Estimation and Confidence Intervals Assignment

## Problem Statement

A manufacturer of print-heads wants to estimate the mean durability of print-heads in terms of the number of characters printed before failure.

A sample of 15 print-heads was tested until failure. The durability values are given in **millions of characters**:

`1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29`

We need to calculate:

1. **99% Confidence Interval using sample standard deviation**
2. **99% Confidence Interval using known population standard deviation = 0.2 million characters**


In [2]:
# Import required libraries
import numpy as np
import scipy.stats as stats
import math

In [3]:
# Given sample data in millions of characters
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85,
                 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

# Sample size
n = len(data)

# Sample mean
sample_mean = np.mean(data)

# Sample standard deviation
# ddof=1 is used because this is sample standard deviation
sample_std = np.std(data, ddof=1)

print("Sample Size (n):", n)
print("Sample Mean:", round(sample_mean, 4))
print("Sample Standard Deviation:", round(sample_std, 4))

Sample Size (n): 15
Sample Mean: 1.2387
Sample Standard Deviation: 0.1932


## A. 99% Confidence Interval Using Sample Standard Deviation

### Why do we use the t-distribution?

We use the **t-distribution** because:

- The sample size is small, only **15 observations**.
- The population standard deviation is **unknown**.
- We are using the **sample standard deviation** as an estimate of the population standard deviation.

### Formula

\[
CI = \bar{x} \pm t_{\alpha/2, n-1} \times \frac{s}{\sqrt{n}}
\]

Where:

- \(\bar{x}\) = sample mean
- \(s\) = sample standard deviation
- \(n\) = sample size
- \(t_{\alpha/2, n-1}\) = critical value from t-distribution
- Confidence level = 99%
- \(\alpha = 1 - 0.99 = 0.01\)
- Degrees of freedom = \(n - 1 = 14\)


In [4]:
# Confidence level
confidence_level = 0.99

# Alpha value
alpha = 1 - confidence_level

# Degrees of freedom
df = n - 1

# t critical value for 99% confidence interval
t_critical = stats.t.ppf(1 - alpha/2, df)

# Margin of error using sample standard deviation
margin_error_t = t_critical * (sample_std / math.sqrt(n))

# Confidence interval
ci_t_lower = sample_mean - margin_error_t
ci_t_upper = sample_mean + margin_error_t

print("Degrees of Freedom:", df)
print("t Critical Value:", round(t_critical, 4))
print("Margin of Error:", round(margin_error_t, 4))
print("99% Confidence Interval using t-distribution:")
print("Lower Limit:", round(ci_t_lower, 4))
print("Upper Limit:", round(ci_t_upper, 4))

Degrees of Freedom: 14
t Critical Value: 2.9768
Margin of Error: 0.1485
99% Confidence Interval using t-distribution:
Lower Limit: 1.0902
Upper Limit: 1.3871


### Interpretation

Using the sample standard deviation, the **99% confidence interval** for the mean durability of the print-heads is approximately:

\[
(1.0902,\ 1.3871)
\]

This means we are **99% confident** that the true mean durability of the print-heads lies between **1.0902 million characters** and **1.3871 million characters**.


## B. 99% Confidence Interval Using Known Population Standard Deviation

Here, the population standard deviation is known:

\[
\sigma = 0.2
\]

Since the population standard deviation is known, we use the **z-distribution**.

### Formula

\[
CI = \bar{x} \pm z_{\alpha/2} \times \frac{\sigma}{\sqrt{n}}
\]

Where:

- \(\bar{x}\) = sample mean
- \(\sigma\) = population standard deviation
- \(n\) = sample size
- \(z_{\alpha/2}\) = z critical value


In [5]:
# Known population standard deviation
population_std = 0.2

# z critical value for 99% confidence interval
z_critical = stats.norm.ppf(1 - alpha/2)

# Margin of error using known population standard deviation
margin_error_z = z_critical * (population_std / math.sqrt(n))

# Confidence interval
ci_z_lower = sample_mean - margin_error_z
ci_z_upper = sample_mean + margin_error_z

print("Population Standard Deviation:", population_std)
print("z Critical Value:", round(z_critical, 4))
print("Margin of Error:", round(margin_error_z, 4))
print("99% Confidence Interval using z-distribution:")
print("Lower Limit:", round(ci_z_lower, 4))
print("Upper Limit:", round(ci_z_upper, 4))

Population Standard Deviation: 0.2
z Critical Value: 2.5758
Margin of Error: 0.133
99% Confidence Interval using z-distribution:
Lower Limit: 1.1057
Upper Limit: 1.3717


### Interpretation

Using the known population standard deviation of **0.2 million characters**, the **99% confidence interval** for the mean durability is approximately:

\[
(1.1057,\ 1.3717)
\]

This means we are **99% confident** that the true mean durability of the print-heads lies between **1.1057 million characters** and **1.3717 million characters**.


In [6]:
# Final summary table
import pandas as pd

summary = pd.DataFrame({
    "Method": [
        "Using Sample Standard Deviation (t-distribution)",
        "Using Known Population Standard Deviation (z-distribution)"
    ],
    "Mean": [sample_mean, sample_mean],
    "Standard Deviation Used": [sample_std, population_std],
    "Critical Value": [t_critical, z_critical],
    "Margin of Error": [margin_error_t, margin_error_z],
    "Lower Limit": [ci_t_lower, ci_z_lower],
    "Upper Limit": [ci_t_upper, ci_z_upper]
})

summary.round(4)

,Method,Mean,Standard Deviation Used,Critical Value,Margin of Error,Lower Limit,Upper Limit
0,Using Sample Standard Deviation (t-distribution),1.2387,0.1932,2.9768,0.1485,1.0902,1.3871
1,Using Known Population Standard Deviation (z-d...,1.2387,0.2000,2.5758,0.1330,1.1057,1.3717


## Final Conclusion

In this assignment, we estimated the average durability of print-heads using confidence intervals.

- When the **population standard deviation was unknown**, the **t-distribution** was used because the sample size was small and the sample standard deviation was used.
- When the **population standard deviation was known**, the **z-distribution** was used.
- The confidence interval using the known population standard deviation is slightly narrower because the population standard deviation is fixed and known.

### Final Answers

1. **99% CI using sample standard deviation:**  
   **(1.0902, 1.3871) million characters**

2. **99% CI using known population standard deviation:**  
   **(1.1057, 1.3717) million characters**
